### DISCUSSION TITLE
paginate + limit

### DISCUSSION THREAD

**Discussion by krthr** [adonisjs/core#4137] [Help]
Hello! Is there any way to limit the total number of rows used for pagination? I don't want to use all the rows in the table for pagination, but instead limit it to the first 1000.

I tried using limit=2 to test the idea, but it didn´t work:

```
      .orderBy('distance', 'desc')
      .limit(2)
      .paginate(page, 30)
```

**Comment by v360tech-sadhana** [comment]
Try chaining these methods:

`.where('id', '>', 0) .andWhere('id', '<', 1001)`

Only for your usecase. There is no other possible solution I guess except this.
